Infrastructure/Factories/TribalSpreadsheetTuitionPartnerFactory.cs

```diff
@@ -0,0 +1,542 @@
+﻿using System.Text.RegularExpressions;
+using Application.Constants;
+using Application.Extensions;
+using Application.Extraction;
+using Application.Factories;
+using Domain;
+using Domain.Constants;
+using Infrastructure.Mapping;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Factories;
+
+public class TribalSpreadsheetTuitionPartnerFactory : ITribalSpreadsheetTuitionPartnerFactory
+{
+    public const string OrganisationDetailsSheetName = "Organisation Details";
+    private const string DeliverySheetName = "Delivery";
+    private const string PricingSheetName = "Pricing";
+    private const int MaxRows = 100000;
+    private const decimal VATRatePercentage = 20;
+
+    private readonly ILogger _logger;
```

Comment:
The logger should be specific to this class: `ILogger<TribalSpreadsheetTuitionPartnerFactory>`

Reply:
Agreed, change made.